### PR TITLE
Port `bin/update-crates` over to Diesel

### PR DESCRIPTION
**Pull request by sgrif**
I thought we were done, but we're not... I forgot the binaries. Sorry.
:(

This one was pretty straightforward. There's a bit of refactoring in
here as well, and I moved some of the integrity checks that this code
was doing to the database. It's worth noting that the new body of
`set_updated_at` is identical to `diesel_set_updated_at`, but I can't
actually call that function from the other since it's `RETURNS trigger`.
Not sure if it's worth resolving, since the function is unlikely to
change.

### CODE REVIEW

**Comment by sgrif** [comment]
Sorry, there's going to be at least 5 more of these. I forgot the `bin/` folder :(

**Comment by carols10cents** [comment]
rustfmt please :) https://travis-ci.org/rust-lang/crates.io/jobs/250921522

**Comment by sgrif** [comment]
Can we either update to the latest version of rustfmt or dump the config file so that the new version won't change the formatting from what we're currently using?

**Comment by sgrif** [comment]
Hm actually even dumping the config doesn't make the newest version have the same output. :\

**Comment by carols10cents** [comment]
I thought we had dumped the config... https://github.com/rust-lang/crates.io/blob/master/.rustfmt.toml

I'm all for updating to the latest version tho

**Comment by sgrif** [comment]
That config is pretty different from what I get by doing `rustfmt --dump-default-config` with the latest version. Doing that doesn't seem to make much of a difference though. I'm guessing that the newest version's differences are either bug fixes or things that aren't specified in the config file

**Comment by sgrif** [comment]
@carols10cents FYI this is marked as waiting on author but I don't think there's anything else waiting on me

**Comment by carols10cents** [comment]
oh sorry

File: src/bin/update-downloads.rs [hunk 1]
```diff
@@ -31,401 +46,335 @@ fn main() {
...
     // Anything older than 24 hours ago will be frozen and will not be queried
     // against again.
     let now = chrono::UTC::now();
     let cutoff = now.naive_utc().date() - chrono::Duration::days(1);
 
-    let mut map = HashMap::new();
-    for row in rows.iter() {
-        let download: VersionDownload = Model::from_row(&row);
-        assert!(map.insert(download.id, download).is_none());
-    }
```
Comment (carols10cents) [inline src/bin/update-downloads.rs]: I am so happy this is getting cleaned up. There was some weird stuff going on in here.

File: src/bin/update-downloads.rs [hunk 2]
```diff
@@ -31,401 +46,335 @@ fn main() {
...
-                             VALUES ($1, $2, $3)",
-                &[&crate_id, &amt, &download.date],
-            )?;
-        }
+        let crate_download = CrateDownload {
+            crate_id: crate_id,
+            downloads: amt,
+            date: download.date,
+        };
+        insert(&crate_download.on_conflict(
```
Comment (carols10cents) [inline src/bin/update-downloads.rs]: i ❤️ upsert

File: src/bin/update-downloads.rs [hunk 3]
```diff
@@ -31,401 +46,335 @@ fn main() {
...
-            None,
-        ).unwrap();
-        let version = Version::insert(
-            &tx,
-            krate.id,
-            &semver::Version::parse("1.0.0").unwrap(),
-            &HashMap::new(),
-            &[],
-        ).unwrap();
-        tx.execute(
```
Comment (carols10cents) [inline src/bin/update-downloads.rs]: \o/

**Pull request merged**